src/tools/idlc/src/plugin.c

```diff
@@ -105,17 +127,34 @@ idlc_load_generator(idlc_generator_plugin_t *plugin, const char *lang)
     }
   }
 
-  if ((handle = openlib(path)) || (lang != path && (handle = openlib(lang)))) {
+  /* open the library */
+  if (lang == path) {
+    handle = openlib(path);
+  } else {
+    handle = openlib(lang);
+    if (handle) {
+      path = lang;
+    } else {
+      handle = openlib(path);
+    }
+  }
+  if (handle) {
     generate = loadsym(handle, "generate");
     if (generate) {
       plugin->handle = handle;
       plugin->generate = generate;
       plugin->generator_options = loadsym(handle, "generator_options");
       plugin->generator_annotations = loadsym(handle, "generator_annotations");
     } else {
+      fprintf(stderr, "no symbol 'generate' found in %s\n", path);
       closelib(handle);
     }
   }
+  else {
+    char errmsg[300];
+    liberror(errmsg, sizeof(errmsg));
+    fprintf(stderr, "can't load %s: %s\n", path, errmsg);
```

Comment:
(**nitpick**) please change to "Cannot load generator %s\n".

Reply:
I'm going to keep both %s: one for the effective name used in loadlib and one for the diagnostic message